### DISCUSSION TITLE
Tasmota, Berry, LVGL on a Lilygo T-Display-S3 Displaying sensor values every 5 seconds problem

### DISCUSSION THREAD

**Discussion by richardsg307** [arendst/Tasmota#20216] [Q&A] [answered]
I've just got Tasmota installed on a Lilygo T-Display-S3 (320x170).  Been messing with the Berry code (new to me) that displays a status line with some text, a clock and the WiFi arc icon. I've also succeeded in getting sensor values from 2 DS18X20 on a breadboard and displayed them centred on the screen every second by using the every_send() driver.

1) I don't know how to change it to every 5 (say) seconds.
2) Each time it displays the new sensor value it doesn't clear the old one so it just becomes a mess. Can't find a 'clear' function.
3) the clock is an hour ahead and I can't find how to change it to UK time.

My use case is to put it on my boiler and have it display outflow/inflow/cabinet temperatures whilst also transmitting them to Home Assistant.
Any help appreciated.

**Comment by sfromis** [comment]
If you want every 5 seconds, you can use `cron` like this:
` tasmota.add_cron('*/5 * * * * *', /->print('ping'), 'pinger-id')`
Obviously instead of that print, use your function similar to what you have in `every_second`.
And with this, you no longer need to add it as a driver, unless you have other reasons for that.
That `pinger-id` is not required, but useful if you want to be able to remove specific `cron` instances.

I'm using HASPmota to manage a similar type display with sensor values, and this works without worrying about lower level details, like you may need when using "raw" LVGL. Without seeing your code, I can't guess what might have worked better.....

To have the local time configured for your place, you need to config the time zone-related settings. I can recommend using this tool to generate a command for the relevant configs. https://tasmotatimezone.com/

**Comment by richardsg307** [comment]
Thanks - I've sorted the 5 seconds with your cron suggestion and got the time right now using the tool you linked to.

Haven't yet sorted the clear text thing yet and still struggling to find any reference to it. To me it seems like a basic thing you would do on a Tasmota ESP with a screen.

The code snippet that writes the sensor values is ...
```
#- Sensor reports -#

def f()
var data = json.load(tasmota.read_sensors())
var oflabel = lv.label(scr)
if f20 != nil oflabel.set_style_text_font(f20, lv.PART_MAIN | lv.STATE_DEFAULT) end
oflabel.set_text(string.format("Outflow = %.2f°C", data['DS18B20-1']['Temperature']))
oflabel.set_style_text_color(lv.color_hex(0xffffff), lv.PART_MAIN)
oflabel.align(lv.ALIGN_CENTER, 0, -45)
var iflabel = lv.label(scr)
if f20 != nil iflabel.set_style_text_font(f20, lv.PART_MAIN | lv.STATE_DEFAULT) end
iflabel.set_text(string.format(" Inflow = %.2f°C", data['DS18B20-2']['Temperature']))
iflabel.set_style_text_color(lv.color_hex(0xffffff), lv.PART_MAIN)
iflabel.align(lv.ALIGN_CENTER, 0, -5)
end

tasmota.add_cron("*/5 * * * * *", f, "every_5_seconds")
```
The pic is taken about 15 seconds after a reboot and the text is going fuzzy already and just gets worse.
![20231213_003608](https://github.com/arendst/Tasmota/assets/67514113/e0f40eec-f1df-4651-b0c5-f204ee59776b)

**Reply by sfromis** [reply]
Well, not using LVGL, I have no "real answers", but comparing to the sample code from the docs, I do notice that you omitted this for a label:
```
stat_line.refr_size()                                                                # new in LVGL8
stat_line.refr_pos()                                                                 # new in LVGL8
```
I do not even know what that actually means, but it looks like it might be relevant. As as small detail, I also noticed that you're using `lv.color_hex(0xffffff)` where the sample code uses `lv.color(0xFFFFFF)`. Again, IDK if that's useful, just something I saw.....

**Comment by richardsg307** [comment, marked as answer]
I went over to HASPmota which I got fairly quickly despite the daunting look of the page file. With that I didn't need to worry about clearing the old values it just did it for me when I updated.
